frontend/src/app/questions/questions.component.html

```diff
@@ -0,0 +1,158 @@
+                <small class="p-error" *ngIf="submitted && (question.title === undefined || !question.title)">
+                    Title is required.
+                </small>
+            </div>
+            <div class="formgrid grid field">
+                <div class="field col-12 md:col-6">
+                    <label for="questionTopics">Topics</label>
+                    <p-multiSelect
+                        required="true"
+                        [style]="{ width: '100%' }"
+                        [options]="topics"
+                        formControlName="selectedTopics"
+                        optionLabel="label"
+                        optionValue="value"
+                        placeholder="Select Topics" />
+                    <small
+                        class="p-error"
+                        *ngIf="submitted && (question.topics === undefined || question.topics.length === 0)">
+                        Topic is required.
+                    </small>
+                </div>
+                <div class="field col-12 md:col-6">
+                    <label for="questionTopics">Difficulty</label>
+                    <p-dropdown
+                        [required]="true"
+                        [style]="{ width: '100%' }"
+                        [options]="difficulties"
+                        formControlName="selectedDifficulty"
+                        optionLabel="label"
+                        optionValue="value"
+                        placeholder="Select Difficulty" />
+                    <small
+                        class="p-error"
+                        *ngIf="submitted && (question.difficulty === undefined || !question.difficulty)">
+                        Difficulty is required.
+                    </small>
+                </div>
+            </div>
+            <div class="field grid col-12" field>
+                <label for="questionDescription">Description</label>
+                <textarea
+                    formControlName="textDescription"
+                    [required]="true"
+                    id="questionDescription"
+                    type="text"
+                    rows="6"
+                    class="text-base text-color surface-overlay p-2 border-1 border-solid surface-border border-round appearance-none outline-none focus:border-primary w-full">
+                </textarea>
+                <small
+                    class="p-error"
+                    *ngIf="submitted && (question.description === undefined || !question.description)">
+                    Description is required.
+                </small>
+            </div>
+        </form>
+
+        <ng-template pTemplate="footer">
+            <p-button label="Cancel" [text]="true" severity="secondary" (onClick)="isDialogVisible = false" />
+            <p-button label="Save" class="p-button-success" (onClick)="saveQuestion()" />
```

Comment:
How about disabling this based on form validators? Something like `[disabled]='!questionFormGroup.valid'`

Reply:
Yup, will do